dev/benchmarks/macrobenchmarks/lib/src/animated_advanced_blend.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+import 'package:flutter/material.dart';
+
+// Various tests to verify that animated image filtered layers do not
+// dirty children even without explicit repaint boundaries. These intentionally use
+// text to ensure we don't measure the opacity peephole case.
+class AnimatedAdvancedBlend extends StatefulWidget {
+  const AnimatedAdvancedBlend({ super.key });
+
+  @override
+  State<AnimatedAdvancedBlend> createState() => _AnimatedBlurBackdropFilterState();
+}
+
+class _AnimatedBlurBackdropFilterState extends State<AnimatedAdvancedBlend> with SingleTickerProviderStateMixin {
+  late final AnimationController controller = AnimationController(vsync: this, duration: const Duration(milliseconds: 5000));
+  late final Animation<double> animation = controller.drive(Tween<double>(begin: 0.0, end: 1.0));
+  ui.ImageFilter imageFilter =
+      const ColorFilter.mode(
+            Colors.black,
+            BlendMode.multiply);
+
+  @override
+  void initState() {
+    super.initState();
+    controller.repeat();
+    animation.addListener(() {
+      setState(() {
+        imageFilter = ColorFilter.mode(
+            Color.fromARGB(255, (animation.value * 255).floor(), 0, 0),
+            BlendMode.multiply);
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        body: Stack(
+          children: <Widget>[
+            ListView(
+              children: <Widget>[
+                for (int i = 0; i < 30; i++)
+                  Center(
+                    child: Transform.scale(scale: 1.01, child: const ModeratelyComplexWidget()),
+                  ),
+              ],
+            ),
+            BackdropFilter(
+              filter: imageFilter,
+              child: const SizedBox.expand(),
+            ),
```

Comment:
@jonahwilliams this PR is backed up in a PR chain.  While I resolve that, can you verify that this is a good test to capture the performance of advanced blends we are interested in?  This basically does the same thing as the animated blur test but instead of blurring it uses a ColorFilter with the multiply blend.

I'm not sure if we have to custom drawing with the blend mode to get an accurate measurement.

Reply:
You absolutely don't want to use a ColorFilter or a BackdropFilter for this. The advanced blend edge case we care about can easily be expressed using a custom painter that draws a solid color rectangle with a blend mode set to an advanced blend. Though it doesn't seem like it, this internally causes the same sort of pass ending cycling as a backdrop filter - unless we have subpass support/framebuffer fetch.